nltk/translate/nist_score.py

```diff
@@ -118,17 +115,15 @@ def corpus_nist(list_of_references, hypotheses, n=5):
     # Calculate corpus-level brevity penalty.
     bp = nist_length_penalty(ref_lengths, hyp_lengths)
 
-    # Collects the various precision values for the different ngram orders.
-    p_n = [Fraction(p_numerators[i], p_denominators[i], _normalize=False)
-           for i, _ in enumerate(range(1,n+1))]
-
     # Eqn 2 in Doddington (2002):
     # Info(w_1 ... w_n) = log_2 [ (# of occurrences of w_1 ... w_n-1) / (# of occurrences of w_1 ... w_n) ]
-    info = [0 if p_n[i].numerator == 0 or p_n[i+1].numerator == 0 # Handles math domain and zero division errors.
-            else math.log(p_n[i].numerator / p_n[i+1].numerator)
-            for i in range(len(p_n)-1)]
-    return sum(info_i/sysoutput_lengths[i] for i, info_i in enumerate(info)) * bp
+    info = [p_numerators[i] / p_numerators[i+1] for i in range(1, len(p_numerators))]
+    # Eqn 3 in Doddington (2002)
+    score = sum(info_i/p_denominators[i] for i, info_i in enumerate(info, start=1)) * bp
```

Comment:
From the paper, in Eqn (3), I'm assuming the `sum(l)` (for all w_1 ... w_n) in sysoutput) to be the hypothesis outputs. But when we do the division, the number of n-grams goes to a really low number. 

The `Info(w_1...w_n)` computation looks correct though.

Reply:
Note that in MT eval, they took the max sysoutput_length instead of the sum: https://github.com/moses-smt/mosesdecoder/blob/master/scripts/generic/mteval-v13a.pl#L888